### PR TITLE
Migrate build from Groovy to Kotlin DSL

### LINKED ISSUE #222: Migrate build from gradle to kotlin
The other java repos all prefer .kts for build scripts. Let's get this one off of groovy and over to kotlin for consistency.

### PR DESCRIPTION
**Pull request by AlexCzar**
Fixes #222

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -13,44 +13,46 @@ plugins {
...
     gradleEnterprise {
         server = gradleEnterpriseServer
+
         buildScan {
-            uploadInBackground = !isCI
+            isUploadInBackground = !isCI
 
-            publishIfAuthenticated()
-            publishAlways()
+            publishAlwaysIf(geAccessKey.isNotEmpty())
```
Comment (AlexCzar) [inline settings.gradle.kts:44]: I might be wrong, but I think there was a typo in the groovy version. If not, I can adjust.

**Comment by breedx-splk** [comment]
Could use a rebase, sorry @AlexCzar

File: settings.gradle.kts (hunk 1 above)
Comment (breedx-splk) [reply settings.gradle.kts:44]: The two blocks definitely makes it harder to read. The differences between the two aren't huge, can they be combined?

**Review by breedx-splk — APPROVED** [review]
Thanks for taking this on!

**Comment by AlexCzar** [comment]
Will probably have to wait for the weekend.

File: settings.gradle.kts (hunk 1 above)
Comment (AlexCzar) [reply settings.gradle.kts:35]: refactored into one block with checks inside

**Comment by AlexCzar** [comment]
Rebased and addressed the comment about the `buildScan` checks.
Please, do not apply any renovate bot changes before merging this.

**Comment by trask** [comment]
> Please, do not apply any renovate bot changes before merging this.

oh no, sorry! I missed this and just merged several

**Comment by trask** [comment]
I will fix merge conflicts...

**Comment by trask** [comment]
thanks @AlexCzar!

**Pull request merged**